generator/src/main.rs

```diff
@@ -162,6 +165,125 @@ impl Parser {
         }
     }
 
+    fn parse_feature(&mut self) {
+        loop {
+            use XmlEvent::*;
+            match self.reader.next().expect("failed to parse XML") {
+                StartElement { name, .. } => match &name.local_name[..] {
+                    "require" => {
+                        self.parse_feature_required();
+                    }
+                    _ => {
+                        eprintln!("unimplemented feature element: {}", name.local_name);
+                        self.finish_element();
+                    }
+                },
+                EndElement { name } => {
+                    if name.local_name == "feature" {
+                        break;
+                    }
+                    eprintln!("unexpected end element: {}", name);
+                }
+                EndDocument => {
+                    panic!("unexpected end of document");
+                }
+                _ => {}
+            }
+        }
+    }
+
+    fn parse_feature_required(&mut self) {
```

Comment:
This looks like it has a lot of copy-paste from `parse_extension_required`. Can we factor that out?

Reply:
I was able to unify them into `parse_required` by adding parameters to optionally pass extension name and number.